core/src/main/java/org/mule/runtime/core/internal/processor/strategy/AbstractCachedThreadReactorSinkProvider.java

```diff
@@ -25,18 +25,39 @@ public abstract class AbstractCachedThreadReactorSinkProvider implements Reactor
 
   private static final int THREAD_CACHE_TIME_LIMIT_IN_MINUTES = 60;
   private static final int TRANSACTION_CACHE_TIME_LIMIT_IN_MINUTES = 10;
+  private boolean sinkIndexEnabled;
 
-  private final Cache<Thread, FluxSink<CoreEvent>> sinks =
+  private final Cache<String, FluxSinkWrapper> sinks =
+      Caffeine.newBuilder()
+          .removalListener((RemovalListener<String, FluxSink<CoreEvent>>) (String, coreEventFluxSink,
+                                                                           removalCause) -> coreEventFluxSink.complete())
+          .expireAfterAccess(THREAD_CACHE_TIME_LIMIT_IN_MINUTES, MINUTES).build();
+
+
+  private final Cache<Thread, FluxSink<CoreEvent>> legacySinks =
       Caffeine.newBuilder().weakKeys()
           .removalListener((RemovalListener<Thread, FluxSink<CoreEvent>>) (thread, coreEventFluxSink,
                                                                            removalCause) -> coreEventFluxSink.complete())
           .expireAfterAccess(THREAD_CACHE_TIME_LIMIT_IN_MINUTES, MINUTES).build();
-  private final Cache<Transaction, FluxSink<CoreEvent>> sinksNestedTx =
+
+  private final Cache<String, FluxSinkWrapper> sinksNestedTx =
+      Caffeine.newBuilder()
+          .removalListener((RemovalListener<String, FluxSink<CoreEvent>>) (transaction, coreEventFluxSink,
+                                                                           removalCause) -> coreEventFluxSink.complete())
+          .expireAfterAccess(TRANSACTION_CACHE_TIME_LIMIT_IN_MINUTES, MINUTES).build();
+
+  private final Cache<Transaction, FluxSink<CoreEvent>> legacySinksNestedTx =
       Caffeine.newBuilder().weakKeys()
           .removalListener((RemovalListener<Transaction, FluxSink<CoreEvent>>) (transaction, coreEventFluxSink,
                                                                                 removalCause) -> coreEventFluxSink.complete())
           .expireAfterAccess(TRANSACTION_CACHE_TIME_LIMIT_IN_MINUTES, MINUTES).build();
 
+  public AbstractCachedThreadReactorSinkProvider() {}
+
+  public AbstractCachedThreadReactorSinkProvider(boolean enabled) {
```

Comment:
use a strategy intead of having this flag queried everywhere

Reply:
I don’t think it makes sense to extract an strategy for something related to internal use in this PR taking into account the difference in used is only due to a kill switch (that should be removed eventually)